docs/insert_sql_userguide.md

```diff
@@ -0,0 +1,130 @@
+# TiSpark Insert SQL User Guide
+
+TiSpark enables you to use TiDB as the data source of Apache Spark, similar to other data sources (PostgreSQL, HDFS, S3, etc.).
+
+It supports insert data using Spark SQL now.
+
+## Use Spark SQL
+1. Initiate `SparkConf`.
+
+    ```scala
+    val sparkConf = new SparkConf().
+      setIfMissing("spark.master", "local[*]").
+      setIfMissing("spark.app.name", getClass.getName).
+      setIfMissing("spark.sql.extensions", "org.apache.spark.sql.TiExtensions").
+      setIfMissing("spark.tispark.pd.addresses", "pd0:2379").
+      setIfMissing("spark.tispark.tidb.addr", "tidb").
+      setIfMissing("spark.tispark.tidb.port", "4000")
+
+    val spark = SparkSession.builder.config(sparkConf).getOrCreate()
+    ```
+
+2. Write using Spark SQL.
+
+    ```scala
+    /* create table before run the code
+    CREATE TABLE tpch_test.target_table_orders (
+      `O_ORDERKEY` int(11) NOT NULL,
+      `O_CUSTKEY` int(11) NOT NULL,
+      `O_ORDERSTATUS` char(1) NOT NULL,
+      `O_TOTALPRICE` decimal(15,2) NOT NULL,
+      `O_ORDERDATE` date NOT NULL,
+      `O_ORDERPRIORITY` char(15) NOT NULL,
+      `O_CLERK` char(15) NOT NULL,
+      `O_SHIPPRIORITY` int(11) NOT NULL,
+      `O_COMMENT` varchar(79) NOT NULL
+    )
+    */
+   
+   //use catalog
+   spark.sql("use tidb_catalog")
+    
+   // write data to tidb
+    spark.sql(
+          "insert into tpch_test.target_table_orders values" +
+            "(1,1,'0',11.11,date'2022-1-1','first','spark',12345,'no comment')")
+
+    // select data
+    val df = spark.sql("select * from tpch_test.target_table_orders")
+   ```
+
+## TiDB Options
+
+The following table shows the TiDB-specific options, which can be passed in through `SparkConf`.
```

Comment:
password can't be passed by SparkConf, it can only be passed by `spark.conf.set`. they are different for the latter one is session level config, and it will include the configs in SparkConf (`SparkContext.getOrCreate().getConf`)
you need to remind user not set password in SparkConf at least. about other configs, they can set by both the SparkConf and `spark.conf.set`